### PR TITLE
Validating whether metadata.json exists for each pattern, and reorganising pattern directories to support this

### PR DESCRIPTION
**Pull request by dastra**
*Description of changes:*

Updated .github/metadata/metadata_json_validator.py to validate whether metadata.json exists for each test pattern directory, and to validate it.  Where it does not or validation fails, comments on the PR.

Reorganised the /*-test-samples/ directories to allow for checks to be made.  Standardised directory structure across languages.

Changes to metadata.json format:

- Twitter and linkedin ids in metadata.json are now the id rather than the full URL.
- pattern_source has an additional type of Partner 
- Added optional pattern_source_name to describe which partner or customer the pattern came from

All metadata.json files comply with this format.

Files that have moved around:

**Documentation**:

- dotnet-test-samples/async-architectures/README.md and img/* have moved to dotnet-test-samples/README-ASYNC.md and img/
- typescript-test-samples/async-architectures/README.md has moved to typescript-test-samples/README-ASYNC.md

**Patterns**:

- dotnet-test-samples/async-architectures/async-lambda-dynamodb has moved to dotnet-test-samples/async-lambda-dynamodb
- dotnet-test-samples/async-architectures/async-lambda-sqs has moved to dotnet-test-samples/async-lambda-sqs
- typescript-test-samples/async-architectures/async-lambda-dynamodb has moved to typescript-test-samples/async-lambda-dynamodb

**Created new metadata.json files for**:

-  java-test-samples/async-lambda-dynamodb/
-  dotnet-test-samples/schema-and-contract-testing/

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dotnet-test-samples/async-lambda-sqs/README.md [hunk 1]
```diff
@@ -6,12 +6,12 @@
 [![test: integration](https://img.shields.io/badge/Test-Integration-yellow)]()
 
 # Asynchronous Integration Test with Amazon Simple Queue Service (SQS)
-You may use a variety of resource types to create the event listener for your asynchronous system under test ([more about event listeners](https://github.com/aws-samples/serverless-test-samples/tree/main/dotnet-test-samples/async-architectures#create-event-listeners-in-test-environments)).
```
Comment (brnkrygs) [inline dotnet-test-samples/async-lambda-sqs/README.md:9]: Can we keep this link in? Maybe needs to point to the README-ASYNC file?

File: typescript-test-samples/appsync-cognito-lambda-aleios/metadata.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   "framework": "CDK",
   "services": ["appsync", "lambda", "cognito"],
   "git_repo_url": "https://github.com/aws-samples/serverless-test-samples",
-  "pattern_source": "aleios",
+  "pattern_source": "Customer",
```
Comment (brnkrygs) [inline typescript-test-samples/appsync-cognito-lambda-aleios/metadata.json:11]: Could this be 'Partner'?

File: typescript-test-samples/launchdarkly-lambda-dynamodb/metadata.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   "framework": "CDK",
   "services": ["lambda", "dynamodb"],
   "git_repo_url": "https://github.com/aws-samples/serverless-test-samples",
-  "pattern_source": "AWS",
+  "pattern_source": "Customer",
```
Comment (brnkrygs) [inline typescript-test-samples/launchdarkly-lambda-dynamodb/metadata.json:11]: Could this be 'Partner'?

**Review by brnkrygs — CHANGES_REQUESTED** [review]
A couple of small comments to check out.

Other thought: moving the async patterns like this may break external links. Do you know if we have a way to track inbound links to see what traffic we may have there? Maybe we should leave at least READMEs at in the old folder structure, with a link to the new spot?

Finally... will the new folder structure break the sync to serverlessland? Or were those patterns just not published over there, *due* to the extra child directory?

File: typescript-test-samples/apigw-lambda-sqs-sns-datadog/metadata.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
   "framework": "CDK",
   "services": ["lambda","sqs","sns","apigw"],
   "git_repo_url": "https://github.com/aws-samples/serverless-test-samples",
-  "pattern_source": "Datadog",
+  "pattern_source": "Customer",
```
[the remark this answers is not in the document]
Comment (dastra) [reply typescript-test-samples/apigw-lambda-sqs-sns-datadog/metadata.json]: Updated type to Partner and added pattern_source_name as Datadog

File: typescript-test-samples/appsync-cognito-lambda-aleios/metadata.json (hunk 2 above)
Comment (dastra) [reply typescript-test-samples/appsync-cognito-lambda-aleios/metadata.json]: Updated type to Partner and added pattern_source_name as Aleios

File: typescript-test-samples/launchdarkly-lambda-dynamodb/metadata.json (hunk 3 above)
Comment (dastra) [reply typescript-test-samples/launchdarkly-lambda-dynamodb/metadata.json]: Updated type to Partner and added pattern_source_name as LaunchDarkly

**Comment by dastra** [comment]
I've checked a popular inbound link site and it doesn't report any inbound links to the async-architectures READMEs which were moved.  And we'll amend the sync to serverlessland to ensure those patterns are still synchronised across OK

**Review by brnkrygs — APPROVED** [review]

**Pull request merged**